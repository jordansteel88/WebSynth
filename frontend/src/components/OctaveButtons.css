.octave {
  border: 2px solid rgb(78, 72, 130);
  border-radius: 3px;
  width: 30%;
  margin: 20px 0 20px 0;
}

.octave h4 {
  margin: -15px auto 10px auto;
  background-color: #282c34;
  width: 30%;
  font-weight: 600;
}

.octave button {
	background-color: #4c527e;
	color: #ccc;
	font-size: 1.1rem;
	line-height: .6;
	text-align: center;
	padding: 8px 12px;
	margin: 0 auto;
  border: 0;
  border-radius: 3px;
	transition: all 0.1s ease-in-out;
}

.octave button:active {
	background-color: #007bff;
}