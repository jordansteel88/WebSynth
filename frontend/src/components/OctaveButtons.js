import React, { useEffect, useState } from 'react';
import './OctaveButtons.css';

const OctaveButtons = ({controls, callback}) => {
  const [octave, setOctave] = useState(controls); 

  useEffect(() => {
    console.log(octave);
    callback(octave)
  }, [callback, octave])

  const clickUp = () => {
    setOctave(octave + 1);
  }  
  
  const clickDown = () => {
    setOctave(octave - 1);
  }

  return (
    <div className="octave">
      <h4>Octave</h4>
        <div className="WaveformForm-radios">
          <button name="down" onClick={clickDown}>-</button>
          <button name="up" onClick={clickUp}>+</button>          
        </div>     
    </div>
  )
}

export default OctaveButtons;