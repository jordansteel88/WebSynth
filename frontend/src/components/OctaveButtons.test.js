import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import OctaveButtons from "./OctaveButtons";

describe('Octave Buttons', () => {

  let setOctave = jest.fn();

  it("button up click should call function", function () {
    const { getByRole } = render(<OctaveButtons 
                                    controls={4} 
                                    callback={setOctave} />);
   
    expect(setOctave).toBeCalledTimes(1);

    let el = getByRole('button', {name: /\+/i});
    userEvent.click(el);
    expect(setOctave).toBeCalledTimes(2);
  });  
  
  it("button down click should call function", function () {
    const { getByRole } = render(<OctaveButtons 
                                    controls={4} 
                                    callback={setOctave} />);
   
    expect(setOctave).toBeCalledTimes(1);

    let el = getByRole('button', {name: /\-/i});
    userEvent.click(el);
    expect(setOctave).toBeCalledTimes(2);
  });

});
