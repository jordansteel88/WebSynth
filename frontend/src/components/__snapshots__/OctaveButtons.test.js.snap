// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Octave Buttons matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="octave"
  >
    <h4>
      Octave
    </h4>
    <div
      class="WaveformForm-radios"
    >
      <button>
        -
      </button>
      <button>
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
